docs/usage/configuration/config_documentation.md

```diff
@@ -3765,6 +3806,109 @@ redis:
   port: 6379
   password: <secret_password>
 ```
+---
+## Worker configuration for workers configuration
+These switches configure each worker them self with the a worker configuration file.
+
+Note also the configuration for the
+[main process above](#worker-configuration-for-main-process).
+
+The manual how to configure workers is described in detail in the
+[worker documentation](../../workers.md)
+
+---
+### `worker_app`
+
+The type of worker. The currently available worker applications are listed
+in [worker documentation](../../workers.md#available-worker-applications).
+
+The most common worker is the
+[`synapse.app.generic_worker`](../../workers.md#synapseappgeneric_worker).
+
+Example configuration:
+```yaml
+worker_app: synapse.app.generic_worker
+```
+---
+### `worker_name`
```

Comment:
Presumably this is required---no default?

We should probably say that worker names need to be unique---things will break if two workers have the same name!

Reply:
There is a default: https://github.com/matrix-org/synapse/blob/0d59ae706a09e95c500a594159fff992260147ac/synapse/config/workers.py#L155
IMO it is only needed if you delegate some tasks to the worker, e.g. `stream_writer`. For a simple `generic_worker` it is not needed.
Related issue: https://github.com/matrix-org/synapse/issues/8084

Reply:
> IMO it is only needed if you delegate some tasks to the worker, e.g. stream_writer.

This seems plausible.

We could maybe say "we strongly recommend giving each worker a unique `worker_name`", but I'm also happy to leave this as it is.

Reply:
=> https://github.com/matrix-org/synapse/pull/14086/commits/c0434bde3b8acbbb283e243fd78e35f85177b1e0